buildlib/azure-pipelines-perf.yml

```diff
@@ -0,0 +1,129 @@
+trigger: none
+pr:
+  branches:
+    include:
+    - master
+    - v*.*.x
+
+variables:
+- name: WorkDir
+  value: /hpc/scrap/azure/$(Build.DefinitionName)/$(Build.BuildId)-$(Build.BuildNumber)
+- name: perf_nodes
+  value: vulcan03,vulcan04
+- name: threshold
+  value: 5
+
+resources:
+  repositories:
+  - repository: PerfX
+    type: github
+    name: Mellanox-lab/PerfX
+    endpoint: Al3xR01
+    ref: master
+
+stages:
+  - stage: Prepare
```

Comment:
Why do you need so many stages here? They depend on one another, in this case you can even put it all into a single job with multiple step. Or I might be missing something here.

Reply:
Separating into stages improves visibility. It's easier to see which stage fails from the summary page, no other benefits.

Reply:
But preparation and cleanup are utility stages, real customers won't benefit from seeing them IMO.

Reply:
Agree, it was easier for me to debug.
Can dense it all into a single stage.
@yosefe WDYT?

Reply:
My personal preference is to keep stages separated for better visibility.